models/PyCryptoBot.py

```diff
@@ -795,7 +795,8 @@ def startApp(self, account, last_action='', banner=True):
                     endDate = startDate + timedelta(minutes=(self.getGranularity()/60)*300)
                     while len(tradingData) != 300 and attempts < 10:
                         tradingData = self.getHistoricalData(self.getMarket(), self.getGranularity(),
-                                                             startDate.isoformat(timespec='milliseconds'))
```

Comment:
If you add a state date it will automatically add 300 intervals from that date without specifying and end date. Are you sure this is a problem?

Reply:
The problem occurs when you use Binance in simulation mode. Line 320 of PyCryptoBot.py is `if iso8601start != '' and iso8601end != '':`. In this case `iso8601end` will always be an empty string.

Reply:
Actually, there are two more instances this needs to be fixed. I'll update the pull request.

Reply:
Thanks, I don't think it is mandatory for the end date for Coinbase Pro and it makes it a lot simpler not including it. If you want to include the end date please make this update for Binance exchange only.